internal/replacements/replacements_test.go

```diff
@@ -290,12 +290,69 @@ var _ = Describe("replace_misconfigured_pods", func() {
 				pod.Spec.NodeSelector = map[string]string{
 					"dummy": "test",
 				}
-				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
 				Expect(needsRemoval).To(BeFalse())
 				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
+		Context("when the securityContext doesn't match", func() {
+			BeforeEach(func() {
+				pClass = fdbv1beta2.ProcessClassStorage
+				remove = false
+			})
+
+			It("should need a removal for RunAsUser change", func() {
```

Comment:
Can we try to better group those different tests together in different `When` statements with their set of `BeforeEach` statements? e.g.


```go
		When("the securityContext doesn't match", func() {
			var needsRemoval bool
			var err error

			BeforeEach(func() {
				pClass = fdbv1beta2.ProcessClassStorage
				remove = false
			})

			JustBeforeEach(func() {
				needsRemoval, err = processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
			})

			When("the last-applied-spec key is updated ", func() {
				BeforeEach(func() {
					pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] = "banana"
				})

				When("the FSGroup is changed", func() {
					BeforeEach(func() {
						pod.Spec.SecurityContext = &corev1.PodSecurityContext{FSGroup: new(int64)}
					})

					It("should need a removal for FSGroup change", func() {
						Expect(needsRemoval).To(BeTrue())
						Expect(err).NotTo(HaveOccurred())
					})
				})

				When("the RunAsUser is changed", func() {
					BeforeEach(func() {
						pod.Spec.SecurityContext = &corev1.PodSecurityContext{RunAsUser: new(int64)}
					})

					It("should need a removal for RunAsUser change", func() {
						Expect(needsRemoval).To(BeTrue())
						Expect(err).NotTo(HaveOccurred())
					})
				})
			})
		})
```